plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/AmazonS3Reference.java

```diff
@@ -32,36 +32,32 @@
 
 package org.opensearch.repositories.s3;
 
-import com.amazonaws.auth.AWSCredentialsProvider;
-import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3Client;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.services.s3.S3Client;
 
 import org.opensearch.common.Nullable;
 import org.opensearch.common.concurrent.RefCountedReleasable;
 
-import java.io.Closeable;
-import java.io.IOException;
-
 /**
- * Handles the shutdown of the wrapped {@link AmazonS3Client} using reference
+ * Handles the shutdown of the wrapped {@link S3Client} using reference
  * counting.
  */
-public class AmazonS3Reference extends RefCountedReleasable<AmazonS3> {
-    AmazonS3Reference(AmazonS3 client) {
+public class AmazonS3Reference extends RefCountedReleasable<S3Client> {
```

Comment:
Since we renamed AmazonS3 to S3Client might make more sense to rename this class to S3ClientReference

Reply:
That rename was in AWS SDK v2, and renaming this one will explode the diff. We can queue some more renaming after.